### PR TITLE
Do not use translations in `EnumServiceProvider`

### LINKED ISSUE #259: Calling translation on boot method on EnumServiceProvider problem
Calling translations on boot method on [EnumServiceProvider](https://github.com/BenSampo/laravel-enum/blob/master/src/EnumServiceProvider.php#L58) cause the invocation of loadJsonPaths function of [Translation FileLoader](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Translation/FileLoader.php#L137). 
This blocks the normal load of the others packages translation files.

I think the best option is to use [Custom Validation Rules](https://laravel.com/docs/9.x/validation#custom-validation-rules)

### PR DESCRIPTION
**Pull request by trippo**
Resolves #259  
Resolves https://github.com/BenSampo/laravel-enum/issues/245

### CODE REVIEW

**Comment by suleymanozev** [comment]
@BenSampo Please review. This issue is causing an error in Nova.

**Comment by daniouyea** [comment]
@BenSampo Please review. This issue is causing an error in Nova.

**Comment by trippo** [comment]
@daniouyea and @suleymanozev i created a package to replace this called [laravel-enum-helper](https://github.com/datomatic/laravel-enum-helper) and  related [laravel-enum-collections](https://github.com/datomatic/laravel-enum-collections) and [nova-enum-field](https://github.com/datomatic/nova-enum-field)

**Comment by spawnia** [comment]
I believe this PR would break the existing translations entirely, so it is not something we can merge and release in the current major version.

**Comment by spawnia** [comment]
I don't understand entirely what changes would be necessary for users of this library to adapt to this change. Can you describe in `UPGRADE.md`?

File: README.md [hunk 1]
```diff
@@ -701,6 +701,25 @@ You can also use the 'pipe' syntax for rules.
 **enum_key**_:enum_class_  
 **enum**_:enum_class_
 
+#### Be carefull
```
Comment (spawnia) [inline README.md:704]:
```suggestion
#### Validation messages
```

File: README.md [hunk 2]
```diff
@@ -701,6 +701,25 @@ You can also use the 'pipe' syntax for rules.
...
+If you want to have validation messages, you must declares them on resources/lang/[locale]/validation.php
```
Comment (spawnia) [inline README.md:705]: How would the app behave if no such messages were defined?

File: composer.json [hunk 3]
```diff
@@ -21,8 +21,8 @@
     "require": {
         "php": "^8.0",
         "composer/composer": "^2.2",
-        "illuminate/contracts": "^9.0",
-        "illuminate/support": "^9.0",
+        "illuminate/contracts": "^9.0|^10.0",
```
Comment (spawnia) [inline composer.json:24]: Laravel 10 support was added in https://github.com/BenSampo/laravel-enum/commit/0d1b8abc32a75dc378fe912c5bf7b2d77323f8fb, you would just need to update your branch with the latest changes in `master`.

**Review by spawnia — CHANGES_REQUESTED** [review]

**Comment by spawnia** [comment]
Closing due to lack of response.

**Pull request closed without merging**